cunumeric/eager.py

```diff
@@ -431,6 +456,57 @@ def diag(self, rhs, extract, k, stacklevel):
             self.array[:] = np.diag(rhs.array, k)
             self.runtime.profile_callsite(stacklevel + 1, False)
 
+    def diag_helper(
+        self,
+        rhs,
+        diag_size,
+        offset,
+        naxes,
+        extract,
+        stacklevel=0,
+        callsite=None,
+    ):
+        if self.shadow:
+            rhs = self.runtime.to_eager_array(rhs, stacklevel=(stacklevel + 1))
+        elif self.deferred is None:
+            self.check_eager_args((stacklevel + 1), rhs)
+        if self.deferred is not None:
+            self.deferred.diag_helper(
+                rhs,
+                diag_size,
+                offset,
+                naxes,
+                extract,
+                stacklevel=(stacklevel + 1),
+            )
+        else:
+            # this should be called only if number of axes <=2
```

Comment:
Based on my understanding so far, this statement is inaccurate. It looks like you will either get `extract == False` and `naxes = 0`, or `extract == True` and `naxes >= 2`

Reply:
Correct, this is a leftover from a different implementation of naxes/extract logic